### ISSUE TITLE
Generate mesh

### ISSUE THREAD

**Issue by Hromon** [WaveEngine/Feedback#54]
How create custom mesh maked in code?
I want generate terrain from perlin noise, but i do not know how I can create a mesh.
![image](https://user-images.githubusercontent.com/29508584/106174463-1fbede80-61a6-11eb-8db6-d35ebec0f95f.png)
![image](https://user-images.githubusercontent.com/29508584/106174525-2f3e2780-61a6-11eb-8233-b4e97fe12541.png)
I must transfer IntPtr from array Vector3?

P.S.
Sorry my english :D

**Comment by Hromon** [comment]
And Inverse kinematic exist in engine? If not, are you going to add it?

**Comment by Hromon** [comment]
Ok, i figured out the mesh.

**Comment by vicfergar** [comment]
Hi @Hromon,


> How create custom mesh maked in code?

Here you have a code snippet of how to instantiate Mesh instance:
```
var graphicsContext = Application.Current.Container.Resolve<GraphicsContext>();
BufferDescription vertexBufferDescription = new BufferDescription((uint)(Marshal.SizeOf(typeof(VertexPositionNormalTangentColorDualTexture)) * this.vertices.Count), BufferFlags.ShaderResource | BufferFlags.VertexBuffer, ResourceUsage.Default);
Buffer vBuffer = graphicsContext.Factory.CreateBuffer(this.vertices.ToArray(), ref vertexBufferDescription);
VertexBuffer vertexBuffer = new VertexBuffer(vBuffer, VertexPositionNormalTangentColorDualTexture.VertexFormat);

BufferDescription indexBufferDescription = new BufferDescription((uint)(sizeof(ushort) * this.indices.Count), BufferFlags.IndexBuffer, ResourceUsage.Default);
Buffer iBuffer = graphicsContext.Factory.CreateBuffer(this.indices.ToArray(), ref indexBufferDescription);
IndexBuffer indexBuffer = new IndexBuffer(iBuffer);

var mesh = new Mesh(new VertexBuffer[] { vertexBuffer }, indexBuffer, PrimitiveTopology.TriangleList);
```


> And Inverse kinematic exist in engine? If not, are you going to add it?

Please, open a new issue for this topic in order to keep the repo organized. Thanks 😄

**Comment by vicfergar** [comment]
So @Hromon as the problem seems to be fixed, can we close the issue?

**Comment by Hromon** [comment]
> So @Hromon as the problem seems to be fixed, can we close the issue?

Yes, thanks, but still I think this method is extremely overloaded with complexity.

**Comment by Hromon** [comment]
And again the question of generation. I do not create new issues so as not to lose the context of the topic.

Am I losing speed with Geometric?

**Comment by vicfergar** [comment]
Hi @Hromon,

I dont really understand your question. In terms of performance, if you are not generating a mesh every time it is fine. Geometric it's only an abstraction over the low level API shown before.

**Comment by Hromon** [comment]
o/

If I need to generate large volumes of meshes, will I lose speed? A striking example of this is chunks in a voxel landscape.

**Comment by vicfergar** [comment]
Hi @Hromon,

Using `Geometry` should not suppose a big difference in speed.
